app/src/main/java/org/wikipedia/analytics/metricsplatform/ArticleEvent.kt

```diff
@@ -138,11 +140,15 @@ class ArticleToolbarInteraction(private val fragment: PageFragment) : TimedMetri
     fun reset() { timer.reset() }
 
     private fun submitEvent(action: String) {
+
         submitEvent(
             "article_toolbar_interaction",
-            mapOf(
-                "action" to action,
-                "time_spent_ms" to timer.elapsedMillis
+            getInteractionData(
+                "article_toolbar_interaction",
+                action,
+                null,
+                "time_spent_ms.$timer.elapsedMillis",
+                null, null, null, null
```

Comment:
Since you have assigned a default `null` values to these parameters, these `null` can be ignored now.

```
getInteractionData(
    "article_toolbar_interaction",
    action,
    null,
    "time_spent_ms.$timer.elapsedMillis"
)
```

Reply:
cool - updated in latest commit